### ISSUE TITLE
Power button, always on (disable power on/off)

### ISSUE THREAD

**Issue by teeriksen** [OlivierZal/com.melcloud#45]
Hi @OlivierZal, great app and use this almost every day 
Appreciate whether a solution can be established to avoid turning off the heat pump. If you accidentally press the power button inside the Melcloud app in Homey, it will take around 15 minutes before the pump runs normally after a wrong press of the power button. A solution to avoid this could e.g. be placed under settings where you can activate/deactivate an 'always on' setting to avoid unexpected power cycle if you accidentally press the power button for the heat pump in the Melcloud app on Homey.

Is it possible to get this established?

**Comment by OlivierZal** [comment]
Hi @teeriksen, you're totally right... that's why the setting exists!

![IMG_1751](https://user-images.githubusercontent.com/88216225/200604250-edb7eedc-534c-46bd-a241-6e1c35ef98fb.PNG)

It's called "Keep Device On": if you don't see it, that means you use the old version of the device. You just need to delete it and re-add it to see the new settings. If you have some flows configured you'll need to make them again but it is once and for all.

Don't hesitate to contact me again for any question or assistance.
Kind regards

**Comment by teeriksen** [comment]
Thanks for quick answer. Looks like my app is outdated regards this. I will try to re-add it to get this feature :-)

**Comment by OlivierZal** [comment]
@teeriksen, don't hesitate to contact me again in case of need!

**Comment by teeriksen** [comment]
Deleted and re-added the app. I cant find simalar settings as your picture states (I guess its under settings?) This is app version 5.6.5 on Homey Pro (early 2019, Homey version 8.0.8)
![IMG_14D77619509E-1](https://user-images.githubusercontent.com/80590127/200611122-965985a4-865f-4e71-8a43-d399af1c258c.jpeg)

**Comment by OlivierZal** [comment]
@teeriksen, maybe I wasn't clear enough: you don't need to remove the app but to remove your device.
Did you perform this action?

**Comment by OlivierZal** [comment]
Oh, you have an air-to-air device.

The setting is only for air-to-water device for now, but will be for air-to-air device very soon.

I reopen the issue as a feature reminder.

**Comment by teeriksen** [comment]
I did try that to, still the same. Yes Uwano 7000 is a air-to-air heat pump device. Perfect

**Comment by OlivierZal** [comment]
@teeriksen, the code is ready (https://github.com/OlivierZal/com.melcloud/pull/46/files) and will be part of the next release.

**Comment by OlivierZal** [comment]
@teeriksen, you can test the feature by downloading this test version: https://homey.app/fr-fr/app/com.mecloud/MELCloud/test/

**Comment by teeriksen** [comment]
Installed based on your app-link on my Homey Pro. When I switch off the app, the app goes dimmed (It looks like that the air-to-air heat pump, Uwano 7000 went off inside the app). I visually did check to see the indoor unit and the heat pump still runs. A little bit confused on this. 

I possible there should instead be a warning to state that power-off is disabled (regards the settings) if you press the button and the app should not go dimmed if the settings is enabled on 'Keep Device On'. Also the explanation could perhaps be more defined. I guess that you can still switch of the unit by its own remote control unit. Example "This settings will deactivate to power off the unit if power button is pressed inside this application"

**Comment by OlivierZal** [comment]
There might be a bug, I’ll investigate.

**Comment by OlivierZal** [comment]
Found, I prepare a fix

**Comment by teeriksen** [comment]
While you're at it, I have a suggestion to simplify/understanding the fan speed on the heat pumps. What if you e.g. could define this corresponding to the menus as the various mode choices. A little easier to see which speed you have chosen
<img width="411" alt="image" src="https://user-images.githubusercontent.com/80590127/201107614-8b0dd96d-41f5-4328-b18f-1d510d8f304d.png">

**Comment by OlivierZal** [comment]
Fixed with the new test version: https://homey.app/fr-fr/app/com.mecloud/MELCloud/test/

Concerning your proposal, it's not against the idea, I'll think if it's possible to do so in a way which doesn't break the existing flows.

**Comment by teeriksen** [comment]
Downloaded & installed updated version. Tried to switch off the device with power button inside the app. The button goes dark (off) and comes on again after 1/2 a second or so. Have only tested based on settings activated 'Keep Device On' Works like a charm :-)

**Comment by OlivierZal** [comment]
I just made a silly code mistake...

Even if it becomes dark, the "off" information is never sent to the physical device.

**Comment by teeriksen** [comment]
> Fixed with the new test version: https://homey.app/fr-fr/app/com.mecloud/MELCloud/test/
> 
> Concerning your proposal, it's not against the idea, I'll think if it's possible to do so in a way which doesn't break the existing flows.

I´ll guess that you set the fan speed based on numbers, so hopefully the GUI part can keep the same fan speed number.

**Comment by OlivierZal** [comment]
Yes, technically I could make it just by transforming the number keys into string keys, but I made so many changes and break flows since I took the app responsibility 1 or 2 month ago that I'm not sure it's the right moment to change the user interface :-) If I do so, the "fan speed" tab will be automatically moved and merged with the other capabilities (vanes & operation modes), because Homey organizes tabs according to capability types, so some users may think they loose in terms of user experience (and others not...).

But I keep the idea somewhere in my mind for later, maybe some day I'll organize a pool via the Homey community to ask for the advice of the 2000+ Homey-MELCloud users.

**Comment by teeriksen** [comment]
> I just made a silly code mistake...
> 
> Even if it becomes dark, the "off" information is never sent to the physical device.

Btw, the log still reports that the device was switched off/on (off=Via Web, On=MELcloud)

**Comment by OlivierZal** [comment]
Yes, because there is the request you made from Homey and the answer you got from the device. 

Between both, the “always on” setting cheats by sending another information (“on”) from the one you requested (“off”). 

Can you confirm that your physical device remained up while you was performing this test? That’s the most important.

**Comment by teeriksen** [comment]
> Yes, technically I could make it just by transforming the number keys into string keys, but I made so many changes and break flows since I took the app responsibility 1 or 2 month ago that I'm not sure it's the right moment to change the user interface :-) If I do so, the "fan speed" tab will be automatically moved and merged with the other capabilities (vanes & operation modes), because Homey organizes tabs according to capability types, so some users may think they loose in terms of user experience (and others not...).
> 
> But I keep the idea somewhere in my mind for later, maybe some day I'll organize a pool via the Homey community to ask for the advice of the 2000+ Homey-MELCloud users.

Fully understand, and great that you take this idea in to future needs. And also, if you ask for 2000+ users advices, you might also get 2000+ different answers.

**Comment by teeriksen** [comment]
> Yes, because there is the request you made from Homey and the answer you got from the device.
> 
> Between both, the “always on” setting cheats by sending another information (“on”) from the one you requested (“off”).
> 
> Can you confirm that your physical device remained up while you was performing this test? That’s the most important.

Answer is yes!

**Comment by OlivierZal** [comment]
Good!
